core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -2515,6 +2515,47 @@ private SqlDialect nonOrdinalDialect() {
         .withPostgresql().ok(expectedPostgresql);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-7056">[CALCITE-7056]
+   * Convert RelNode to Sql failed when the RelNode includes quantify operators</a>.
+   */
+  @Test void testQuantifyOperatorsWithTypeCoercion() {
+    final String query = "SELECT timestamp '1970-01-01 01:23:45'"
```

Comment:
What would be the output if the input SQL is the same as that described in Jira?

Reply:
It would be ok, and I had changed the case as JIRA description:
```
    final String query = "SELECT '1970-01-01 01:23:45'"
        + " = any (array[timestamp '1970-01-01 01:23:45',"
        + "timestamp '1970-01-01 01:23:46'])";
``` @NobiGo